src/coreclr/jit/llvm.cpp

```diff
@@ -2013,7 +2091,20 @@ void Llvm::ConvertShadowStackLocalNode(GenTreeLclVarCommon* node)
             blk->SetLayout(_compiler->typGetObjLayout(handle));
             blk->gtBlkOpKind = GenTreeBlk::BlkOpKindInvalid;
         }
-        CurrentRange().InsertBefore(node, offset, shadowStackLocal, lclAddress);
+
+
+        if (indirOper == GT_NONE)
+        {
+            // for GT_LCL_VAR_ADDR, the GT_ADD the result of the conversion,
+            // so replace node with the GT_ADD
+            node->ReplaceWith(lclAddress, _compiler);
+
+            CurrentRange().InsertBefore(node, offset, shadowStackLocal);
+        }
+        else
+        {
+            CurrentRange().InsertBefore(node, offset, shadowStackLocal, lclAddress);
+        }
```

Comment:
How about this:
```suggestion
        CurrentRange().InsertBefore(node, shadowStackLocal, offset);

        if (indirOper == GT_NONE)
        {
            // Local address nodes are directly replaced with the ADD.
            node->ReplaceWith(lclAddress, _compiler);
        }
        else
        {
            CurrentRange().InsertBefore(node, lclAddress);
        }
```
Reads a little cleaner to me.

Reply:
Yes, this looks better